core/dispatcher/src/main/scala/whisk/core/container/ContainerPool.scala

```diff
@@ -633,8 +638,33 @@ class ContainerPool(
     }
 }
 
-object ContainerPool {
+object ContainerPool extends Logging {
     def requiredProperties = Map(selfDockerEndpoint -> "localhost") ++ Map(dockerImageTag -> "latest") ++ Map(invokerContainerNetwork -> "bridge") ++ Map(invokerContainerPolicy -> "")
     type RunResult = (Instant, Instant, Option[(Int, String)])
-    val defaultMaxActive = 4
+
+    private val defaultNumCore = 4  // The number of cores if it cannot be determined dynamically with nproc
+    // The shareFactor indicates the number of containers that would share a single core, on average.
+    // cpuShare is a docker option (-c) whereby a container's CPU access is limited.
+    // A value of 1024 is the full share so a strict resource division with a shareFactor of 2 would yield 512.
+    // On an idle/underloaded system, a container will still get to use underutilized CPU shares.
+    val shareFactor = 2
+    val cpuShare = (1024.0 / getDefaultMaxActive()).toInt
+
+    /*
+     * The total number of containers is simply the number of cores dilated by the cpu sharing.
+     */
+    def getDefaultMaxActive() = getNumberOfCore() * ContainerPool.shareFactor
+
+    /*
+     * Retrieve the number of cores on the host machine - defaulting to defaultNumCore if unknown.
+     */
+    private def getNumberOfCore(): Int = try {
+        val (stdout, stderr, exitCode) = SimpleExec.syncRunCmd(Array("/usr/bin/nproc"))(TransactionId.invoker)
```

Comment:
Thanks for the detailed clarification. Given how unreliable all methods are, I support the opinion expressed by others to make this part of the config. I think it's acceptable (at least as a first step) to assume homogeneous invokers and specify the number of cores only once? (As opposed to: `invoker.0.numcores = .., invoker.1.numcores = ...`)